### PR TITLE
Added tent construct/deconstruct activities

### LINKED ISSUE #40599: Setting up and taking down tents takes no time
# Describe the bug

Setting up and taking down a tent takes no time.

# Steps To Reproduce

1. find a tent (campsite start)
2. examine middle tile to deconstruct
3. notice that no time has passed
4. activate to set it up again
5. repeate step 3

# Expected behavior

I know  that setting up and taking down a tent should take at least some time, due to prior experience with camping and incompetent people.

# Screenshots

setting up:
before
after

taking down:
before
after

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-2157-gf1e2374 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Alternative Map Key [alt_map_key],
    Stats Through Skills [StatsThroughSkills]
]

### PR DESCRIPTION
**Pull request by Jamuro-g**
#### Summary
Bugfixes "Added tent construct&deconstruct activities"

#### Purpose of change
Deploying a tent as well as taking it down didn't have activities set for them and didn't take time (deconstruct only took 2 seconds)
This change gives both actions an activity (currently set to take 20 minutes ... but i am open to suggestions)

Fixes #40599

#### Describe the solution

- Added tent_placement_activity_actor & tent_deconstruct_activity_actor, called from iuse (placement) and iexamine (deconstruct)

#### Testing

- Spawn in a tent and large tent
- Deploy them
- Deconstruct both

### CODE REVIEW

**Comment by actual-nh** [comment]
Ideally, one would have a JSON field(s) in the item's entry that would say how long it would take (on average), with 20 or whatever as a default. However, whatever the minimum amount of time for any of the tents is would be better than currently.

File: src/activity_actor.cpp [hunk 1]
```diff
@@ -2905,7 +3010,7 @@ std::unique_ptr<activity_actor> play_with_pet_activity_actor::deserialize( JsonI
     JsonObject data = jsin.get_object();
 
     data.read( "pet_name", actor.pet_name );
-
+
```
Comment (kevingranade) [inline src/activity_actor.cpp:3013]:
```suggestion

```

File: src/activity_actor_definitions.h [hunk 2]
```diff
@@ -1045,6 +1080,7 @@ class play_with_pet_activity_actor : public activity_actor
 
         void start( player_activity &act, Character & ) override;
         void do_turn( player_activity &, Character & ) override {}
+
```
Comment (kevingranade) [inline src/activity_actor_definitions.h:1083]:
```suggestion
```

File: src/activity_actor.cpp [hunk 3]
```diff
@@ -2852,6 +2852,111 @@ std::unique_ptr<activity_actor> disassemble_activity_actor::deserialize( JsonIn
...
+    JsonObject data = jsin.get_object();
+    data.read( "moves_total", actor.moves_total );
+    data.read( "wall", actor.wall );
+    data.read( "floor", actor.floor );
+    data.read( "floor_center", actor.floor_center );
+    data.read( "door_closed", actor.door_closed );
+    data.read( "radius", actor.radius );
+    data.read( "it", actor.it );
+    data.read( "target", actor.target );
+
```
Comment (kevingranade) [inline src/activity_actor.cpp:2918]:
```suggestion

```

**Pull request merged**